lib/models/ui/linear_segmented_hsv_colormap.dart

```diff
@@ -0,0 +1,84 @@
+import "package:collection/collection.dart";
+import "package:flutter/foundation.dart";
+import "package:flutter/painting.dart";
+
+/// A linear segmented HSV colormap, just like python's matplotlib.colors.LinearSegmentedColormap.
+/// It is given a map of stops and colors, and it can interpolate between them. For instance,
+/// if we know have the stop 50 with colour blue and the stop 100 with colour red, asking the value
+/// at 75 will yield purple.
+/// Any value below all the stops will yield colour of the smallest stop, and any value above all the stops
+/// will yield the colour of the largest stop.
+@immutable
+class LinearSegmentedHSVColormap {
+  late final List<int> stops;
+  late final List<HSVColor> colors;
+
+  /// Creates a linear segmented HSV colormap.
+  /// [colorStops] is a map of stops and colours (the colour that we want
+  /// to have at each stop). At other values, the colour is interpolated.
+  /// There must be at least one color stop.
+  LinearSegmentedHSVColormap(Map<int, HSVColor> colorsStops) {
+    final unsortedEntries = colorsStops.entries;
+    final sortedEntries = unsortedEntries.sorted(
+      // We sort by stop value (the key of the map)
+      (a, b) => a.key.compareTo(b.key),
+    );
+
+    // Stops are distinct since we use a map
+    stops = sortedEntries.map((entry) => entry.key).toList();
+    colors = sortedEntries.map((entry) => entry.value).toList();
+
+    assert(stops.isNotEmpty, "There must be at least one color stop");
+  }
+
+  /// Creates a linear segmented HSV colormap with a uniform distribution of colors.
+  /// [stops] is a list of values that define the stops of the colormap. Their corresponding
+  /// colors are interpolated between [hueStart] and [hueEnd] (in degrees), with a fixed
+  /// [saturation] and [value]. The distance between each stop colour is constant.
+  /// There must be at least two colour stops.
+  factory LinearSegmentedHSVColormap.uniform(
+    List<int> stops, {
+    double hueStart = 0,
+    double hueEnd = 330,
+    double saturation = 0.8,
+    double value = 0.5,
```

Comment:
I don't think that this variables name is explicit enough. We could maybe rename it or be more precise in the comment above. What do you think?

Reply:
I'm used to working with the HSV colour space, so I might be a bit biased indeed. However, I'm really unsure of how to name those variables in a more explicit manner. Hue, saturation and value are the name of the coordinates in this space (like red, green, blue or x, y, z). 
What other variable name would you consider?

Reply:
Ok so let's keep it like that then!